acl/authorizer.go

```diff
@@ -161,6 +162,280 @@ type Authorizer interface {
 
 	// Embedded Interface for Consul Enterprise specific ACL enforcement
 	enterpriseAuthorizer
+
+	// ToAllowAuthorizer is needed until we can use ResolveResult in all the places this interface is used.
+	ToAllowAuthorizer() AllowAuthorizer
+}
```

Comment:
Eventually this will be unified with the ACLResolveResult structure, but want to take that on as a separate line item. https://github.com/hashicorp/consul/issues/12481